tony-history-server/buildAndDeploy.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+###########################################################################################################
+# NAME: buildAndDeploy.sh
+#
+# DESCRIPTION:
+# This script will send the playBinary.zip generated after running `gradle dist` to host and user given
+# given as inputs.
+#
+#
+# INPUT:
+# $1 - user
+# $2 - hostname
+#
+#
+# OUTPUT:
+# $DEPLOY_LOG: log file that includes all tasks.
+#
+#
+# ENVIRONMENT VARIABLES:
+#
+#
+# NOTES:
+# Run this script in tony-history-server/ folder.
+#
+#
+# EXIT CODE:
+# 0 - Success
+# 1 - Cleaning step failed
+# 2 - Bundling distribution zip step failed
+# 3 - Copying to remote host failed
+#
+#
+# CHANGELOG:
+# OCT 24 2018 PHAT TRAN
+############################################################################################################
+DEPLOY_LOG=deploy.log
+
+echo "Cleaning up old builds..." | tee $DEPLOY_LOG
+gradle clean 2>&1 | tee -a $DEPLOY_LOG
```

Comment:
I believe this can be simplified to:

```
gradle clean |& tee -a $DEPLOY_LOG
```

Reply:
Do you have a reference for this? This syntax looks unfamiliar to me...

Reply:
See https://stackoverflow.com/a/37085215/1128392

Reply:
`|&` is just syntactic sugar for `2>&1 |`

Reply:
👍

Reply:
For some reasons, I couldn't use the `|&` in my script.
```
Cleaning up old builds...
./buildAndDeploy.sh: line 39: syntax error near unexpected token `&'
./buildAndDeploy.sh: line 39: `gradle clean |& tee -a $DEPLOY_LOG'
```

Reply:
I'm going to revert back to the old version of the script (`2>&1` instead of `|&`) and fixed the tab/space indent issue in the next commit.

Reply:
Most likely you have an older version of Bash installed on your Mac (me too, actually). According to the SO post I linked above, `|&` was introduced in Bash 4.0. On my Mac, I have 3.2.57, and `|&` doesn't work either. But once I upgraded bash (`brew install bash`), it worked. (Also have to change `#!/bin/bash` to `#!/usr/bin/env bash` so the new `bash` gets picked up)